subsys/bluetooth/controller/hci_driver.c

```diff
@@ -38,9 +42,13 @@ void blectlr_assertion_handler(const char *const file, const u32_t line)
 
 static int cmd_handle(struct net_buf *cmd)
 {
-	const bool pkt_put = hci_cmd_packet_put(cmd->data);
+	s32_t errcode = MULTITHREADING_LOCK_ACQUIRE();
 
-	if (!pkt_put) {
+	if (!errcode) {
+		errcode = hci_cmd_put(cmd->data);
+		MULTITHREADING_LOCK_RELEASE();
+	}
+	if (errcode) {
 		return -ENOBUFS;
```

Comment:
Should we return errcode instead?
Also I'm wondering about the use of the different errcode types. ble_controller now returns error codes from nrf_errno.h, which is s32_t, while the driver uses int and errno.h
Is it safe to just forward the errcode, or do we need to convert.
At the moment we are using a mix between the two. If it is safe to just forward we should do so in all places.

Reply:
The BLE controller is slightly ambiguous to what error it's experiencing and will return -EINVAL no matter what error it sees. I checked `host/hci_core.c` and saw that it ignores the type of error returned from the HCI driver, so it should be safe to just return whatever the controller returns. In the future, the controller should distinguish between -EINVAL and -ENOBUFS itself.